get_box_mesh.py

```diff
@@ -0,0 +1,14 @@
+def get_box_mesh(dim,a,b,n, t=None, periodic=None):
+    if periodic is None:
+        periodic = (False)*dim
+        
```

Comment:
```suggestion
    if np.isscalar(a):
        a = (a,)*dim
    if np.isscalar(b):
        b = (b,)*dim
    if np.isscalar(n):
        n = (n,)*dim

```
Promoting the scalars to tuples is probably simpler than having the conditional call to `gen` below.